dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx

```diff
@@ -252,9 +253,17 @@ export function JsonRpcContextProvider({
           },
         });
 
-        const valid = EthTransaction.fromSerializedTx(
-          signedTx as any
-        ).verifySignature();
+        let valid = false;
+        const [, reference] = chainId.split(":");
+        if (reference === "44787" || reference === "42220") {
+          // Verify Celo transaction
+          const [, signer] = recoverTransaction(signedTx);
+          valid = signer.toLowerCase() === address.toLowerCase();
+        } else {
```

Comment:
When would we hit this `else` case if Celo testnet/mainnet are covered above? Does it even make sense to keep this logic if it doesn't actually work as expected as you indicated?

Reply:
maybe i misunderstand your question, but i think the `else` statement works for ethereum? i didn't want to touch the existing logic because i don't have the full context, i only noticed it wasn't working on Celo so wanted to fix that :) note that i am using a celo lib to recover the tx, which i'm not sure will work for other chains

Reply:
Oh yea woops my bad. Should've expanded the diff more, for some reason I thought this was a Celo specific function 🤦 

All good sorry for the confusion!